database-plugins/src/main/java/co/cask/hydrator/plugin/DBUtils.java

```diff
@@ -230,6 +234,57 @@ public static DriverCleanup ensureJDBCDriverIsAvailable(Class<? extends Driver>
     return type;
   }
 
+  @Nullable
+  public static Object transformValue(int sqlColumnType, ResultSet resultSet, String fieldName) throws SQLException {
+    Object original = resultSet.getObject(fieldName);
+    if (original != null) {
+      switch (sqlColumnType) {
+        case Types.SMALLINT:
+        case Types.TINYINT:
+          return ((Number) original).intValue();
+        case Types.NUMERIC:
+        case Types.DECIMAL:
+          return ((BigDecimal) original).doubleValue();
+        case Types.DATE:
+          return resultSet.getDate(fieldName).getTime();
+        case Types.TIME:
+          return resultSet.getTime(fieldName).getTime();
+        case Types.TIMESTAMP:
+          return resultSet.getTimestamp(fieldName).getTime();
+        case Types.BLOB:
+          Object toReturn;
+          Blob blob = (Blob) original;
+          try {
+            toReturn = blob.getBytes(1, (int) blob.length());
+          } finally {
+            blob.free();
+          }
+          return toReturn;
+        case Types.CLOB:
+          String s;
+          StringBuilder sbf = new StringBuilder();
+          Clob clob = (Clob) original;
+          try {
+            try (BufferedReader br = new BufferedReader(clob.getCharacterStream(1, (int) clob.length()))) {
```

Comment:
Why so complicated?? Isn't it just `clob.getSubString(1, (int) clob.length())`??